### PR TITLE
perf(trie): use available pre-loaded prefixsets

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

`StateRoot::incremental_root_*` methods internally create a `PrefixSetLoader` which walks the updated accounts and storages within the block range, hashes their keys and constructs corresponding prefixsets. In `insert_hashes` and `get_or_take_block_and_execution_range::<true>` provider methods, this results in double work hidden behind the incremental root abstraction, since we already walk and hash the same keys in order to update hashing tables.

### CODE REVIEW

File: crates/storage/provider/src/traits/hashing.rs [hunk 1]
```diff
@@ -2,28 +2,50 @@ use auto_impl::auto_impl;
...
 /// Hashing Writer
 #[auto_impl(&, Arc, Box)]
 pub trait HashingWriter: Send + Sync {
-    /// Unwind and clear account hashing
-    fn unwind_account_hashing(&self, range: RangeInclusive<BlockNumber>) -> Result<()>;
+    /// Unwind and clear account hashing.
+    ///
+    /// # Returns
+    ///
+    /// Set of hashed keys of updated accounts.
```
Comment (mattsse) [inline crates/storage/provider/src/traits/hashing.rs:17]: I see, returning everything that's updated makes perfect sense

**Comment by rkrasiuk** [comment]
I would like to test this before merging.

**Review by mattsse — APPROVED** [review]
> this results in double work hidden behind the incremental root abstraction, since we already walk and hash the same keys in order to update hashing tables.

nice find, lgtm

**Review by shekhirin — APPROVED** [review]

**Comment by rkrasiuk** [comment]
tested on sepolia, been successfully syncing live for a couple days

**Pull request merged**